engine/src/flutter/lib/ui/window/viewport_metrics.cc

```diff
@@ -65,10 +64,63 @@ ViewportMetrics::ViewportMetrics(
       physical_display_features_state(p_physical_display_features_state),
       display_id(p_display_id) {}
 
+ViewportMetrics::ViewportMetrics(
```

Comment:
Could you expand why we're introducing a new constructor? How bad would it be to add the new arguments to the existing constructor and update all call sites to pass these new values?

Reply:
New constructor was just to avoid having to change all the different platform call sites.  I know Android has one and Fuchsia has another.

Reply:
How many call sites are we talking? I'd prefer to do the right thing unless it's too much.

Reply:
A few.  But where do we draw the line? if I changed SetViewportMetrics in platform_view_android_jni_impl.cc do I also change FlutterJNI.java and FlutterRenderer.java and FlutterView.java - and then we're basically in the platform implementation which Louise was trying to avoid.